contracts/token/ERC20/ERC20Snapshot.sol

```diff
@@ -38,12 +50,23 @@ abstract contract ERC20Snapshot is ERC20 {
     // Snapshot ids increase monotonically, with the first value being 1. An id of 0 is invalid.
     Counters.Counter private _currentSnapshotId;
 
+    /**
+     * @dev Emitted by {_snapshot} when a snapshot identified by `id` is created.
+     */
     event Snapshot(uint256 id);
 
     /**
-     * @dev Creates a new snapshot id. Balances are only stored in snapshots on demand: unless a snapshot was taken, a
-     * balance change will not be recorded. This means the extra added cost of storing snapshotted balances is only paid
-     * when required, but is also flexible enough that it allows for e.g. daily snapshots.
+     * @dev Creates a new snapshot and returns its snapshot id.
+     *
+     * Emits a {Snapshot} event that contains the same id.
+     *
+     * {_snapshot} is internal so that you can decide if and how to expose it externally. If it is exposed, it can be
+     * guarded by something like {AccessControl} or open to the public. The latter is required for implementing certain
+     * trust minimization mechanisms such as forking, but you must consider that it can potentially be used by attackers in
+     * two ways. First, it can be used to increase the cost of retrieval of values from snapshots, although it will grow
+     * logarithmically thus rendering this attack ineffective in the long term. Second, it can be used to target specific
+     * accounts and increase the cost of ERC20 transfers for them, in the ways specified in the Gas Costs secion above.
+     * We haven't measured the actual numbers; if this is something you're interested in please reach out to us.
```

Comment:
```suggestion
     * {_snapshot} is `internal`: you must decide how to expose it externally. This can be done both by
     * guarding it with a system such as {AccessControl}, or by leaving it open to the public. 
     *
     * [NOTE]
     * ====
     * While an open way of calling {_snapshot} is required for certain minimization mechanisms
     * such as forking, you must consider that it can potentially be used by attackers in
     * two ways. 
     *
     * First, it can be used to increase the cost of retrieval of values from snapshots, although it will grow
     * logarithmically thus rendering this attack ineffective in the long term. Second, it can be used to target specific
     * accounts and increase the cost of ERC20 transfers for them, in the ways specified in the Gas Costs section above.
     *
     * We haven't measured the actual numbers; if this is something you're interested in please reach out to us.
     * ====
```

Reply:
I tried to reduce the wall-of-textyness: https://deploy-preview-2186--openzeppelin-contracts-docs.netlify.com/contracts/2.x/api/token/erc20#ERC20Snapshot-_snapshot--